File: 461. Hamming Distance/461. Hamming Distance.js
/**
 * @param {number} x
 * @param {number} y
 * @return {number}
 */
var hammingDistance = function(x, y) {
    var z = x^y;
    res = 0;
    while(z) {
        if(z%2 === 1) res++;
        z = parseInt(z / 2);
    }
    return res;
};

File: 461. Hamming Distance/461. Hamming Distance.rb
# @param {Integer} x
# @param {Integer} y
# @return {Integer}
def hamming_distance(x, y)
    z = x ^ y
    res = 0
    while z != 0
       res += 1 if z%2 == 1

       z = z / 2
    end
    res
end
